hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/TokenAttributesValidator.java

```diff
@@ -57,6 +64,36 @@ public TokenAttributesValidator() {
         // Dagger
     }
 
+    /**
+     * Performs a "pure" check for a given account ID. If the account ID is not in the body, is null, or is invalid, a {@link PreCheckException} is thrown.
+     */
+    public static void validateAccountId(final boolean bodyHasAccountId, @Nullable final AccountID accountID)
+            throws PreCheckException {
+        validateTruePreCheck(bodyHasAccountId, INVALID_ACCOUNT_ID);
+        validateAccountID(accountID, INVALID_ACCOUNT_ID);
+    }
+
+    /**
+     * Performs a "pure" check for a given token ID. If the token ID is not in the body, is null, or is invalid, a {@link PreCheckException} is thrown.
+     */
+    public static void validateTokenId(final boolean bodyHasTokenId, @Nullable final TokenID tokenID)
+            throws PreCheckException {
+        validateTruePreCheck(bodyHasTokenId, INVALID_TOKEN_ID);
+        validateTruePreCheck(tokenID != null, INVALID_TOKEN_ID);
+        validateNullableTokenId(tokenID);
+    }
+
+    /**
+     * Performs validation checks for a token ID <b>only if</b> the tokenID is not null.
+     */
+    public static void validateNullableTokenId(@Nullable final TokenID tokenID) throws PreCheckException {
+        if (tokenID != null) {
```

Comment:
Why do we need not null check ?

Reply:
In the case where `tokenID` is null, this method won't throw any NPE's

Reply:
I see. Are we using `validateNullableTokenId` outside `validateTokenId` ? If so, we can keep that. Otherwise we can make this private and make it @NonNull . What do you think ?

Reply:
I removed the validateNullable... method entirely. It's not necessary.